library/tasks/HISAT2.wdl

```diff
@@ -304,3 +310,94 @@ task HISAT2InspectIndex {
     File log_file ="hisat2_inspect.log"
   }
 }
+
+task HISAT2RSEMSingleEnd {
+  File hisat2_ref
+  File fastq
+  String ref_name
+  String output_basename
+  String sample_name
+
+  # runtime values
+  String docker = "quay.io/humancellatlas/secondary-analysis-hisat2:v0.2.2-2-2.1.0"
+  Int machine_mem_mb = 15000
+  Int cpu = 4
+  Int disk = ceil((size(fastq, "GB")) * 100 + size(hisat2_ref, "GB") * 2 + 10)
+  Int preemptible = 5
+  Int max_retries = 0
+
+  meta {
+    description: "This HISAT2 alignment task will align paired-end fastq reads to transcriptome only. "
+  }
+
+  parameter_meta {
+    hisat2_ref: "HISAT2 reference"
+    fastq: "gz fastq file"
+    ref_name: "the basename of the index for the reference genome"
+    output_basename: "basename used for output files"
+    sample_name: "sample name of input"
+    docker: "(optional) the docker image containing the runtime environment for this task"
+    machine_mem_mb: "(optional) the amount of memory (MB) to provision for this task"
+    cpu: "(optional) the number of cpus to provision for this task"
+    disk: "(optional) the amount of disk space (GB) to provision for this task"
+    preemptible: "(optional) if non-zero, request a pre-emptible instance and allow for this number of preemptions before running the task on a non preemptible machine"
+    max_retries: "(optional) retry this number of times if task fails -- use with caution, see skylab README for details"
+  }
+
+  command {
+    set -e
+
+    # fix names if necessary.
+    if [ "${fastq}" != *.fastq.gz ]; then
+        FQ=${fastq}.fastq.gz
+        mv ${fastq} ${fastq}.fastq.gz
+    else
+        FQ=${fastq}
+    fi
+
+
+    tar --no-same-owner -xvf "${hisat2_ref}"
+
+    # increase gap alignment penalty to avoid gap alignment
+    # --mp 1,1 --np 1 --score-min L,0,-0.1 is default paramesters when rsem runs alignment by using bowtie2/Bowtie
+    # --mp 1,1 and --np 1 will reduce mismatching penalty to 1 for all.
+    # with no-splice-alignment no-softclip no-mixed options on, HISAT2 will only output concordant alignment without soft-cliping
+    # --rdg 99999999,99999999 and --rfg 99999999,99999999 will give an infinity penalty to alignment with indel.As results
```

Comment:
```suggestion
    # --rdg 99999999,99999999 and --rfg 99999999,99999999 set an infinite penalty to alignments with indels. 
    # As a result, alignments with gaps or deletions are excluded. 
```
Fixed a copy/pasted typo, ended up rewriting the sentence to make sense.

Reply:
Thanks fixed.